sklearn/linear_model/_base.py

```diff
@@ -691,9 +691,9 @@ def fit(self, X, y, sample_weight=None):
             return_mean=True,
         )
 
-        if sample_weight is not None:
-            # Sample weight can be implemented via a simple rescaling.
-            X, y = _rescale_data(X, y, sample_weight)
+        # Sample weight can be implemented via a simple rescaling.
+        sample_weight = np.sqrt(sample_weight)
```

Comment:
what is the reason for not calling `square_root=True` in the `_rescale_data` and omit this line?

Reply:
Is it because we always want to have the square root of the sample weights available?

Reply:
Yes it's needed after to rescale the X_offset